core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala

```diff
@@ -106,6 +106,28 @@ case class SparkListenerExecutorRemoved(time: Long, executorId: String, reason:
   extends SparkListenerEvent
 
 @DeveloperApi
+case class SparkListenerExecutorBlacklisted(
+    time: Long,
+    executorId: String,
+    taskFailures: Int)
+  extends SparkListenerEvent
+
+@DeveloperApi
+case class SparkListenerExecutorUnblacklisted(time: Long, executorId: String)
+  extends SparkListenerEvent
+
+@DeveloperApi
+case class SparkListenerNodeBlacklisted(
+    time: Long,
+    nodeId: String,
```

Comment:
We've been using "node" and "host" interchangeably in this change set. To be more clear, I'll write this as hostId here and in SparkListenerNodeUnblacklisted.